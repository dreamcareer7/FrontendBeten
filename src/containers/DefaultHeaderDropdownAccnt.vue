<template>
  <CDropdown
    nav
    no-caret
    no-popper
    add-menu-classes="dropdown-menu-right"
  >
    <template #toggler-content>
      <img
        src="img/avatars/6.jpg"
        class="img-avatar"
        alt="admin@bootstrapmaster.com"
      />
    </template>
    <CDropdownHeader tag="div" class="text-center">
      <strong>Account</strong>
    </CDropdownHeader>
    <CDropdownItem>
      <i class="cui-bell"/> Updates
      <CBadge variant="info" additional-classes="hehe">{{ itemsCount }}</CBadge>
    </CDropdownItem>
    <CDropdownItem>
      <i class="cui-envelope-open" /> Messages
      <CBadge variant="success">{{ itemsCount }}</CBadge>
    </CDropdownItem>
    <CDropdownItem>
      <i class="cui-task" /> Tasks
      <CBadge variant="danger">{{ itemsCount }}</CBadge>
    </CDropdownItem>
    <CDropdownItem>
      <i class="cui-comment-square" /> Comments
      <CBadge variant="warning">{{ itemsCount }}</CBadge>
    </CDropdownItem>
    <CDropdownHeader
     tag="div"
      class="text-center"
    >
      <strong>Settings</strong>
    </CDropdownHeader>
    <CDropdownItem>
      <i class="cui-user" /> Profile
    </CDropdownItem>
    <CDropdownItem>
      <i class="cui-wrench" /> Settings
    </CDropdownItem>
    <CDropdownItem>
      <i class="cui-dollar" /> Payments
      <CBadge variant="secondary">{{ itemsCount }}</CBadge>
    </CDropdownItem>
    <CDropdownItem>
      <i class="cui-file" /> Projects
      <CBadge variant="primary">{{ itemsCount }}</CBadge>
    </CDropdownItem>
    <CDropdownDivider/>
    <CDropdownItem>
      <i class="cui-shield" /> Lock Account
    </CDropdownItem>
    <CDropdownItem>
      <i class="cui-lock-locked" /> Logout
    </CDropdownItem>
  </CDropdown>
</template>

<script>
export default {
  name: 'DefaultHeaderDropdownAccnt',
  data () {
    return { itemsCount: 42 }
  }
}
</script>
