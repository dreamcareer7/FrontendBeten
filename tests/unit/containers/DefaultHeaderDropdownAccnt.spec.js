import Vue from 'vue'
import CoreuiVue from '@coreui/vue'
import DefaultHeaderDropdownAccnt from '@/containers/DefaultHeaderDropdownAccnt'
import { shallowMount } from '@vue/test-utils';

Vue.use(CoreuiVue)

describe('DefaultHeaderDropdownAccnt.vue', () => {
  it('has a name', () => {
    expect(DefaultHeaderDropdownAccnt.name).toMatch('DefaultHeaderDropdownAccnt')
  })
  it('has a created hook', () => {
    expect(typeof DefaultHeaderDropdownAccnt.data).toMatch('function')
  })
  it('sets the correct default data', () => {
    expect(typeof DefaultHeaderDropdownAccnt.data).toMatch('function')
    const defaultData = DefaultHeaderDropdownAccnt.data()
    expect(defaultData.itemsCount).toBe(42)
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(DefaultHeaderDropdownAccnt)
    expect(wrapper.element).toMatchSnapshot()
  })
})
