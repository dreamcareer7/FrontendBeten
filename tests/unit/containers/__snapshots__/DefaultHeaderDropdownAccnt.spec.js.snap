// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultHeaderDropdownAccnt.vue renders correctly 1`] = `
<cdropdown-stub
  addmenuclasses="c-dropdown-menu-right"
  nav="true"
  nocaret="true"
  nopopper="true"
  placement="bottom-start"
  togglertext="Dropdown"
>
   
  <cdropdownheader-stub
    class="c-text-center"
    tag="div"
  >
    <strong>
      Account
    </strong>
  </cdropdownheader-stub>
   
  <cdropdownitem-stub
    activeclass="c-active"
    event="click"
    exactactiveclass="c-active"
    routertag="a"
    target="_self"
  >
    <i
      class="cui-bell"
    />
     Updates
    
    <cbadge-stub
      activeclass="c-active"
      additional-classes="hehe"
      event="click"
      exactactiveclass="c-active"
      routertag="a"
      tag="span"
      target="_self"
      variant="info"
    >
      42
    </cbadge-stub>
  </cdropdownitem-stub>
   
  <cdropdownitem-stub
    activeclass="c-active"
    event="click"
    exactactiveclass="c-active"
    routertag="a"
    target="_self"
  >
    <i
      class="cui-envelope-open"
    />
     Messages
    
    <cbadge-stub
      activeclass="c-active"
      event="click"
      exactactiveclass="c-active"
      routertag="a"
      tag="span"
      target="_self"
      variant="success"
    >
      42
    </cbadge-stub>
  </cdropdownitem-stub>
   
  <cdropdownitem-stub
    activeclass="c-active"
    event="click"
    exactactiveclass="c-active"
    routertag="a"
    target="_self"
  >
    <i
      class="cui-task"
    />
     Tasks
    
    <cbadge-stub
      activeclass="c-active"
      event="click"
      exactactiveclass="c-active"
      routertag="a"
      tag="span"
      target="_self"
      variant="danger"
    >
      42
    </cbadge-stub>
  </cdropdownitem-stub>
   
  <cdropdownitem-stub
    activeclass="c-active"
    event="click"
    exactactiveclass="c-active"
    routertag="a"
    target="_self"
  >
    <i
      class="cui-comment-square"
    />
     Comments
    
    <cbadge-stub
      activeclass="c-active"
      event="click"
      exactactiveclass="c-active"
      routertag="a"
      tag="span"
      target="_self"
      variant="warning"
    >
      42
    </cbadge-stub>
  </cdropdownitem-stub>
   
  <cdropdownheader-stub
    class="c-text-center"
    tag="div"
  >
    <strong>
      Settings
    </strong>
  </cdropdownheader-stub>
   
  <cdropdownitem-stub
    activeclass="c-active"
    event="click"
    exactactiveclass="c-active"
    routertag="a"
    target="_self"
  >
    <i
      class="cui-user"
    />
     Profile
  
  </cdropdownitem-stub>
   
  <cdropdownitem-stub
    activeclass="c-active"
    event="click"
    exactactiveclass="c-active"
    routertag="a"
    target="_self"
  >
    <i
      class="cui-wrench"
    />
     Settings
  
  </cdropdownitem-stub>
   
  <cdropdownitem-stub
    activeclass="c-active"
    event="click"
    exactactiveclass="c-active"
    routertag="a"
    target="_self"
  >
    <i
      class="cui-dollar"
    />
     Payments
    
    <cbadge-stub
      activeclass="c-active"
      event="click"
      exactactiveclass="c-active"
      routertag="a"
      tag="span"
      target="_self"
      variant="secondary"
    >
      42
    </cbadge-stub>
  </cdropdownitem-stub>
   
  <cdropdownitem-stub
    activeclass="c-active"
    event="click"
    exactactiveclass="c-active"
    routertag="a"
    target="_self"
  >
    <i
      class="cui-file"
    />
     Projects
    
    <cbadge-stub
      activeclass="c-active"
      event="click"
      exactactiveclass="c-active"
      routertag="a"
      tag="span"
      target="_self"
      variant="primary"
    >
      42
    </cbadge-stub>
  </cdropdownitem-stub>
   
  <cdropdowndivider-stub
    tag="div"
  />
   
  <cdropdownitem-stub
    activeclass="c-active"
    event="click"
    exactactiveclass="c-active"
    routertag="a"
    target="_self"
  >
    <i
      class="cui-shield"
    />
     Lock Account
  
  </cdropdownitem-stub>
   
  <cdropdownitem-stub
    activeclass="c-active"
    event="click"
    exactactiveclass="c-active"
    routertag="a"
    target="_self"
  >
    <i
      class="cui-lock-locked"
    />
     Logout
  
  </cdropdownitem-stub>
</cdropdown-stub>
`;
